src/rtcp/app.cc

```diff
@@ -53,10 +53,12 @@ rtp_error_t uvgrtp::rtcp::handle_app_packet(uint8_t *packet, size_t size)
     }
 
     /* Deallocate previous frame from the buffer if it exists, it's going to get overwritten */
-    if (participants_[frame->ssrc]->app_frame) {
-        delete[] participants_[frame->ssrc]->app_frame->payload;
-        delete   participants_[frame->ssrc]->app_frame;
-    }
+	if (participants_[frame->ssrc]) {
+		delete[] participants_[frame->ssrc]->app_frame->payload;
+		delete   participants_[frame->ssrc]->app_frame;
+	}
```

Comment:
Couple of issues here. Firstly, the indentation is inconsistent with the rest of the code base, please try to use the same code style around the code you're editing. In this case, indent by 4, no tabs.

Secondly, I think the check in `if` is not correct. Now the code checks if there exists a participant with this SSRC and if so, it frees the APP packet memory even if it doesn't exist (it could point to garbage).

What  you must do instead, is to revert the change you made to this `if` and instead add following block of code before the `if`: 
```
    if (!is_participant(frame->ssrc)) {
        LOG_WARN("Got an APP packet from an unknown participant");
        add_participant(frame->ssrc);
    }
```